[{"authorTime":"2018-01-20 05:49:27","codes":[{"authorDate":"2018-01-20 05:49:27","commitOrder":1,"curCode":"    public boolean startSite2SiteVpn(final Site2SiteVpnConnection conn, final VirtualRouter router) throws ResourceUnavailableException {\n        if (router.getState() != State.Running) {\n            s_logger.warn(\"Unable to apply site-to-site VPN configuration, virtual router is not in the right state \" + router.getState());\n            throw new ResourceUnavailableException(\"Unable to apply site 2 site VPN configuration,\" + \" virtual router is not in the right state\", DataCenter.class,\n                    router.getDataCenterId());\n        }\n\n        return applySite2SiteVpn(true, router, conn);\n    }\n","date":"2018-01-20 05:49:27","endLine":618,"groupId":"1905","id":1,"instanceNumber":1,"isCurCommit":0,"methodName":"startSite2SiteVpn","params":"(finalSite2SiteVpnConnectionconn@finalVirtualRouterrouter)","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-cloudstack-10-0.7/blobInfo/CC_OUT/blobs/d2/2dcbafbba540598fe666ca3cd41d71ccd9bb48.src","preCode":"    public boolean startSite2SiteVpn(final Site2SiteVpnConnection conn, final VirtualRouter router) throws ResourceUnavailableException {\n        if (router.getState() != State.Running) {\n            s_logger.warn(\"Unable to apply site-to-site VPN configuration, virtual router is not in the right state \" + router.getState());\n            throw new ResourceUnavailableException(\"Unable to apply site 2 site VPN configuration,\" + \" virtual router is not in the right state\", DataCenter.class,\n                    router.getDataCenterId());\n        }\n\n        return applySite2SiteVpn(true, router, conn);\n    }\n","realPath":"server/src/main/java/com/cloud/network/router/VpcVirtualNetworkApplianceManagerImpl.java","repoName":"cloudstack","snippetEndLine":0,"snippetStartLine":0,"startLine":610,"status":"B"},{"authorDate":"2018-01-20 05:49:27","commitOrder":1,"curCode":"    public boolean startRemoteAccessVpn(final RemoteAccessVpn vpn, final VirtualRouter router) throws ResourceUnavailableException {\n        if (router.getState() != State.Running) {\n            s_logger.warn(\"Unable to apply remote access VPN configuration, virtual router is not in the right state \" + router.getState());\n            throw new ResourceUnavailableException(\"Unable to apply remote access VPN configuration,\" + \" virtual router is not in the right state\", DataCenter.class,\n                    router.getDataCenterId());\n        }\n\n        final Commands cmds = new Commands(Command.OnError.Stop);\n        _commandSetupHelper.createApplyVpnCommands(true, vpn, router, cmds);\n\n        try {\n            _agentMgr.send(router.getHostId(), cmds);\n        } catch (final OperationTimedoutException e) {\n            s_logger.debug(\"Failed to start remote access VPN: \", e);\n            throw new AgentUnavailableException(\"Unable to send commands to virtual router \", router.getHostId(), e);\n        }\n        Answer answer = cmds.getAnswer(\"users\");\n        if (!answer.getResult()) {\n            s_logger.error(\"Unable to start vpn: unable add users to vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \"\n                    + router.getInstanceName() + \" due to \" + answer.getDetails());\n            throw new ResourceUnavailableException(\"Unable to start vpn: Unable to add users to vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId()\n            + \" on domR: \" + router.getInstanceName() + \" due to \" + answer.getDetails(), DataCenter.class, router.getDataCenterId());\n        }\n        answer = cmds.getAnswer(\"startVpn\");\n        if (!answer.getResult()) {\n            s_logger.error(\"Unable to start vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \" + router.getInstanceName() + \" due to \"\n                    + answer.getDetails());\n            throw new ResourceUnavailableException(\"Unable to start vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \"\n                    + router.getInstanceName() + \" due to \" + answer.getDetails(), DataCenter.class, router.getDataCenterId());\n        }\n\n        return true;\n    }\n","date":"2018-01-20 05:49:27","endLine":758,"groupId":"1905","id":2,"instanceNumber":2,"isCurCommit":0,"methodName":"startRemoteAccessVpn","params":"(finalRemoteAccessVpnvpn@finalVirtualRouterrouter)","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-cloudstack-10-0.7/blobInfo/CC_OUT/blobs/d2/2dcbafbba540598fe666ca3cd41d71ccd9bb48.src","preCode":"    public boolean startRemoteAccessVpn(final RemoteAccessVpn vpn, final VirtualRouter router) throws ResourceUnavailableException {\n        if (router.getState() != State.Running) {\n            s_logger.warn(\"Unable to apply remote access VPN configuration, virtual router is not in the right state \" + router.getState());\n            throw new ResourceUnavailableException(\"Unable to apply remote access VPN configuration,\" + \" virtual router is not in the right state\", DataCenter.class,\n                    router.getDataCenterId());\n        }\n\n        final Commands cmds = new Commands(Command.OnError.Stop);\n        _commandSetupHelper.createApplyVpnCommands(true, vpn, router, cmds);\n\n        try {\n            _agentMgr.send(router.getHostId(), cmds);\n        } catch (final OperationTimedoutException e) {\n            s_logger.debug(\"Failed to start remote access VPN: \", e);\n            throw new AgentUnavailableException(\"Unable to send commands to virtual router \", router.getHostId(), e);\n        }\n        Answer answer = cmds.getAnswer(\"users\");\n        if (!answer.getResult()) {\n            s_logger.error(\"Unable to start vpn: unable add users to vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \"\n                    + router.getInstanceName() + \" due to \" + answer.getDetails());\n            throw new ResourceUnavailableException(\"Unable to start vpn: Unable to add users to vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId()\n            + \" on domR: \" + router.getInstanceName() + \" due to \" + answer.getDetails(), DataCenter.class, router.getDataCenterId());\n        }\n        answer = cmds.getAnswer(\"startVpn\");\n        if (!answer.getResult()) {\n            s_logger.error(\"Unable to start vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \" + router.getInstanceName() + \" due to \"\n                    + answer.getDetails());\n            throw new ResourceUnavailableException(\"Unable to start vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \"\n                    + router.getInstanceName() + \" due to \" + answer.getDetails(), DataCenter.class, router.getDataCenterId());\n        }\n\n        return true;\n    }\n","realPath":"server/src/main/java/com/cloud/network/router/VpcVirtualNetworkApplianceManagerImpl.java","repoName":"cloudstack","snippetEndLine":0,"snippetStartLine":0,"startLine":726,"status":"B"}],"commitId":"893a88d225276e45f12f9490e6af2c94a81c2965","commitMessage":"@@@CLOUDSTACK-10105: Use maven standard project structure in all projects (#2283)\n\nRemove maven standard module (which only a few were using) and get ride of maven customization for the projects structure.\n\n- moved all directories to src/main/java.  src/main/resources.  src/main/scripts.  src/test/java.  src/test/resources\n- grep scan to search for src/com and src/org left over\n- grep for <project>/scripts to fix pom.xml configuration\n- remove custom <build> configuration in pom.xml\n\nSigned-off-by: Marc-Aur?le Brothier <m@brothier.org>","date":"2018-01-20 05:49:27","modifiedFileCount":"1","status":"B","submitter":"Marc-Aur?le Brothier"},{"authorTime":"2018-04-26 21:03:46","codes":[{"authorDate":"2018-01-20 05:49:27","commitOrder":2,"curCode":"    public boolean startSite2SiteVpn(final Site2SiteVpnConnection conn, final VirtualRouter router) throws ResourceUnavailableException {\n        if (router.getState() != State.Running) {\n            s_logger.warn(\"Unable to apply site-to-site VPN configuration, virtual router is not in the right state \" + router.getState());\n            throw new ResourceUnavailableException(\"Unable to apply site 2 site VPN configuration,\" + \" virtual router is not in the right state\", DataCenter.class,\n                    router.getDataCenterId());\n        }\n\n        return applySite2SiteVpn(true, router, conn);\n    }\n","date":"2018-01-20 05:49:27","endLine":618,"groupId":"101813","id":3,"instanceNumber":1,"isCurCommit":0,"methodName":"startSite2SiteVpn","params":"(finalSite2SiteVpnConnectionconn@finalVirtualRouterrouter)","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-cloudstack-10-0.7/blobInfo/CC_OUT/blobs/d2/2dcbafbba540598fe666ca3cd41d71ccd9bb48.src","preCode":"    public boolean startSite2SiteVpn(final Site2SiteVpnConnection conn, final VirtualRouter router) throws ResourceUnavailableException {\n        if (router.getState() != State.Running) {\n            s_logger.warn(\"Unable to apply site-to-site VPN configuration, virtual router is not in the right state \" + router.getState());\n            throw new ResourceUnavailableException(\"Unable to apply site 2 site VPN configuration,\" + \" virtual router is not in the right state\", DataCenter.class,\n                    router.getDataCenterId());\n        }\n\n        return applySite2SiteVpn(true, router, conn);\n    }\n","realPath":"server/src/main/java/com/cloud/network/router/VpcVirtualNetworkApplianceManagerImpl.java","repoName":"cloudstack","snippetEndLine":0,"snippetStartLine":0,"startLine":610,"status":"N"},{"authorDate":"2018-04-26 21:03:46","commitOrder":2,"curCode":"    public boolean startRemoteAccessVpn(final RemoteAccessVpn vpn, final VirtualRouter router) throws ResourceUnavailableException {\n        if (router.getState() != State.Running) {\n            s_logger.warn(\"Unable to apply remote access VPN configuration, virtual router is not in the right state \" + router.getState());\n            throw new ResourceUnavailableException(\"Unable to apply remote access VPN configuration,\" + \" virtual router is not in the right state\", DataCenter.class,\n                    router.getDataCenterId());\n        }\n\n        final Commands cmds = new Commands(Command.OnError.Stop);\n        _commandSetupHelper.createApplyVpnCommands(true, vpn, router, cmds);\n\n        try {\n            _agentMgr.send(router.getHostId(), cmds);\n        } catch (final OperationTimedoutException e) {\n            s_logger.debug(\"Failed to start remote access VPN: \", e);\n            throw new AgentUnavailableException(\"Unable to send commands to virtual router \", router.getHostId(), e);\n        }\n        Answer answer = cmds.getAnswer(\"users\");\n        if (answer == null || !answer.getResult()) {\n            String errorMessage = (answer == null) ? \"null answer object\" : answer.getDetails();\n            s_logger.error(\"Unable to start vpn: unable add users to vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \"\n                    + router.getInstanceName() + \" due to \" + errorMessage);\n            throw new ResourceUnavailableException(\"Unable to start vpn: Unable to add users to vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId()\n            + \" on domR: \" + router.getInstanceName() + \" due to \" + errorMessage, DataCenter.class, router.getDataCenterId());\n        }\n        answer = cmds.getAnswer(\"startVpn\");\n        if (answer == null || !answer.getResult()) {\n            String errorMessage = (answer == null) ? \"null answer object\" : answer.getDetails();\n            s_logger.error(\"Unable to start vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \" + router.getInstanceName() + \" due to \"\n                    + errorMessage);\n            throw new ResourceUnavailableException(\"Unable to start vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \"\n                    + router.getInstanceName() + \" due to \" + errorMessage, DataCenter.class, router.getDataCenterId());\n        }\n\n        return true;\n    }\n","date":"2018-04-26 21:03:46","endLine":760,"groupId":"101813","id":4,"instanceNumber":2,"isCurCommit":0,"methodName":"startRemoteAccessVpn","params":"(finalRemoteAccessVpnvpn@finalVirtualRouterrouter)","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-cloudstack-10-0.7/blobInfo/CC_OUT/blobs/ea/bfb4337f4ef4b0d719f7b3888abcac98188bc0.src","preCode":"    public boolean startRemoteAccessVpn(final RemoteAccessVpn vpn, final VirtualRouter router) throws ResourceUnavailableException {\n        if (router.getState() != State.Running) {\n            s_logger.warn(\"Unable to apply remote access VPN configuration, virtual router is not in the right state \" + router.getState());\n            throw new ResourceUnavailableException(\"Unable to apply remote access VPN configuration,\" + \" virtual router is not in the right state\", DataCenter.class,\n                    router.getDataCenterId());\n        }\n\n        final Commands cmds = new Commands(Command.OnError.Stop);\n        _commandSetupHelper.createApplyVpnCommands(true, vpn, router, cmds);\n\n        try {\n            _agentMgr.send(router.getHostId(), cmds);\n        } catch (final OperationTimedoutException e) {\n            s_logger.debug(\"Failed to start remote access VPN: \", e);\n            throw new AgentUnavailableException(\"Unable to send commands to virtual router \", router.getHostId(), e);\n        }\n        Answer answer = cmds.getAnswer(\"users\");\n        if (!answer.getResult()) {\n            s_logger.error(\"Unable to start vpn: unable add users to vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \"\n                    + router.getInstanceName() + \" due to \" + answer.getDetails());\n            throw new ResourceUnavailableException(\"Unable to start vpn: Unable to add users to vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId()\n            + \" on domR: \" + router.getInstanceName() + \" due to \" + answer.getDetails(), DataCenter.class, router.getDataCenterId());\n        }\n        answer = cmds.getAnswer(\"startVpn\");\n        if (!answer.getResult()) {\n            s_logger.error(\"Unable to start vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \" + router.getInstanceName() + \" due to \"\n                    + answer.getDetails());\n            throw new ResourceUnavailableException(\"Unable to start vpn in zone \" + router.getDataCenterId() + \" for account \" + vpn.getAccountId() + \" on domR: \"\n                    + router.getInstanceName() + \" due to \" + answer.getDetails(), DataCenter.class, router.getDataCenterId());\n        }\n\n        return true;\n    }\n","realPath":"server/src/main/java/com/cloud/network/router/VpcVirtualNetworkApplianceManagerImpl.java","repoName":"cloudstack","snippetEndLine":0,"snippetStartLine":0,"startLine":726,"status":"M"}],"commitId":"4c42aafae0d3286a38d796a7c69a6aec6810cf79","commitMessage":"@@@[CLOUDSTACK-10356] Fix NPE in Cloudstack found with NPEDetector  (#2573)\n\n* fix https://issues.apache.org/jira/browse/CLOUDSTACK-10356\n\n* del patch file\n\n* Update ResourceCountDaoImpl.java\n\n* fix some format\n\n* fix code\n\n* fix error message in VolumeOrchestrator\n\n* add check null stmt\n\n* del import unuse class\n\n* use BooleanUtils to check Boolean\n\n* fix error message\n\n* delete unuse function\n\n* delete the deprecated function  updateDomainCount\n\n* add error log and throw exception in ProjectManagerImpl.java\n","date":"2018-04-26 21:03:46","modifiedFileCount":"11","status":"M","submitter":"brett"}]
